src/test/java/StringManipulationTest.java

```diff
@@ -1,116 +1,197 @@
     }
 
     @Test
+    //testRemoveNthCharacter3 calls removeNthCharacter() with a standard String parameter
+	//and parameter n = 1 and ensures the method returns an empty String.
     public void testRemoveNthCharacter3() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("This is a nice sentence.");
+        String NthCharRemoved = manipulatedstring.removeNthCharacter(1, false);
+        assertEquals("", NthCharRemoved);
     }
 
     @Test
+    //testRemoveNthCharacter4 calls removeNthCharacter with a standard String parameter
+    //and parameter n = stringLength and ensures the method removes only the last character.
     public void testRemoveNthCharacter4() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("Remove the last character only");
+        int stringLength = manipulatedstring.getString().length();
+        String NthCharRemoved = manipulatedstring.removeNthCharacter(stringLength, false);
+        assertEquals("Remove the last character onl", NthCharRemoved);
     }
 
     @Test
+    //testRemoveNthCharacter5 calls removeNthCharacter with a standard String parameter
+    //and parameter n = 2 and ensures the method removes and replaces every other character with a space.
     public void testRemoveNthCharacter5() {
-        fail("Not yet implemented");
+    	manipulatedstring.setString("This is a nice sentence.");
+    	String NthCharRemoved = manipulatedstring.removeNthCharacter(2, true);
+    	assertEquals("T i   s a n c   e t n e ", NthCharRemoved);
     }
-
+    
     @Test
+    //testRemoveNthCharacter6 calls removeNthCharacter with n <= 0 and ensures the method throws an IllegalArgumentException
     public void testRemoveNthCharacter6() {
-        fail("Not yet implemented");
+    	assertThrows(IllegalArgumentException.class,
+                ()->{
+                	manipulatedstring.setString("I'd b3tt3r put s0me d161ts in this 5tr1n6, right?");
+                	manipulatedstring.removeNthCharacter(-99, false);
+                });
     }
 
     @Test
+    //testRemoveCharacter7 calls removeNthCharacter with n > stringLength and ensures the method
+    //throws an IndexOutOfBoundsException
     public void testRemoveNthCharacter7() {
-        fail("Not yet implemented");
+    	assertThrows(IndexOutOfBoundsException.class,
+                ()->{
+                	manipulatedstring.setString("I'd b3tt3r put s0me d161ts in this 5tr1n6, right?");
+                	manipulatedstring.removeNthCharacter(99, true);
+                });
     }
 
     @Test
+    //testGeSubStrings1 calls testGeSubStrings with a standard String parameter and ensures the method
+    //returns the correct output
     public void testGeSubStrings1() {
         manipulatedstring.setString("This is my string");
         String [] sStings = manipulatedstring.getSubStrings(3, 4);
-
         assertEquals(sStings[0], "my");
         assertEquals(sStings[1], "string");
     }
 
     @Test
```

Comment:
This whitespace test might have been better suited by a lower out-of-bounds exception test for a negative start word parameter.